src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/prisoners/PrisonersResourceIntTest.kt

```diff
@@ -22,6 +24,122 @@ class PrisonersResourceIntTest : IntegrationTestBase() {
   @Autowired
   private lateinit var nomisDataBuilder: NomisDataBuilder
 
+  @AfterEach
+  fun cleanUp() {
+    repository.deleteOffenders()
+  }
+
+  @Test
+  fun `DSL setup check - latest booking on root`() {
+    lateinit var rootOffender: Offender
+    lateinit var aliasOffender1: Offender
+    lateinit var aliasOffender2: Offender
+    lateinit var activeBooking: OffenderBooking
+    lateinit var oldBooking: OffenderBooking
+    lateinit var aliasBooking: OffenderBooking
+    lateinit var oldAliasBooking: OffenderBooking
+
+    nomisDataBuilder.build {
+      rootOffender = offender(nomsId = "A1234TT") {
+        activeBooking = booking {}
+        oldBooking = booking(active = false)
+
+        aliasOffender1 = alias(lastName = "ALIAS1") {
+          aliasBooking = booking(active = false)
+          oldAliasBooking = booking(active = false)
+        }
+        aliasOffender2 = alias(lastName = "ALIAS2")
+      }
+    }
```

Comment:
The data produced by nomisDataBuilder isn’t correct… I guess that’s why you then load them from the database? That feels like a hack to work around the fact they’re not right. For example aliasOffender1.allBookings is empty before loading from the repository.

I think after the call to nomisDataBuilder any data produced should be in a correct state - maybe that means that the repository load is done within the builders? But expecting people to reload from the database isn’t right.

Reply:
I think the allBookings list being empty for aliases is correct - it is the case when loaded from the database too. The many to one is on the root offender id so only the root offender has bookings in the list. Thats why the getAllBookings() function is needed. And so i think the records returned from the dsls are correct too but i'll add assertions to the tests for that.